packages/core/src/getStateFromPath.tsx

```diff
@@ -289,23 +290,34 @@ const matchAgainstConfigs = (remaining: string, configs: RouteConfig[]) => {
     if (match) {
       const matchedParams = config.pattern
         ?.split('/')
-        .filter((p) => p.startsWith(':'))
+        .map((p, x) => [p, x] as const)
+        .filter(([p]) => p.startsWith(':'))
         .reduce<Record<string, any>>(
-          (acc, p, i) =>
+          (acc, [p, x], i) =>
             Object.assign(acc, {
               // The param segments appear every second item starting from 2 in the regex match result
-              [p]: match![(i + 1) * 2].replace(/\//, ''),
+              // Also include position in the key to handle the same param appearing multiple times
+              [p + x]: match![(i + 1) * 2].replace(/\//, ''),
```

Comment:
Including the index in the key seems problematic.

If path is `foo1` and index is `1`, how would it be distinguished from `foo` and index `11`? While it's an unlikely case, it's still a problematic implementation.

Something like `[p]: { [index]: match![(i + 1) * 2].replace(/\//, '') }` may be more appropriate.

Also use `index` for index name since `x` isn't obvious.

The filter can probably be removed by adding this inside the `reduce` itself, and by adjusting the code accordingly:

```js
if (!p.startsWith(':')) {
  return acc;
}
```

Reply:
I love this suggestion! I will update accordingly.

Reply:
I have made the changes for this. After removing the `filter()`, I noticed we could also get rid of the `map()` as well, and so I took care of that for us!

Thanks for the suggestion. With this approach we are able to do all the processing in one pass 🙌